prebuilt-tasks/src/test/java/com/redhat/parodos/tasks/notification/NotificationWorkFlowTaskTest.java

```diff
@@ -0,0 +1,203 @@
+				"test-subject", null, Arrays.asList("test-group-1", "test-group-2"));
+		putParamsToCtx(dto, ctx);
+
+		WorkReport result = underTest.execute(ctx);
+
+		assertEquals(WorkStatus.COMPLETED, result.getStatus());
+		verify(apiInstanceMock, Mockito.times(1)).create(dto);
+	}
+
+	@Test
+	@SneakyThrows
+	public void testExecuteMissingGroupNamesSuccess() {
+		NotificationMessageCreateRequestDTO dto = buildNotificationMessageCreateRequestDTO("test-type", "test-body",
+				"test-subject", Arrays.asList("test-user-1", "test-user-2"), null);
+		putParamsToCtx(dto, ctx);
+
+		WorkReport result = underTest.execute(ctx);
+
+		assertEquals(WorkStatus.COMPLETED, result.getStatus());
+		verify(apiInstanceMock, Mockito.times(1)).create(dto);
+	}
+
+	@Test
+	@SneakyThrows
+	public void testExecuteMissingUsernamesAndGroupNamesErr() {
+		NotificationMessageCreateRequestDTO dto = buildNotificationMessageCreateRequestDTO("test-type", "test-body",
+				"test-subject", null, null);
+		putParamsToCtx(dto, ctx);
+
+		WorkReport result = underTest.execute(ctx);
+
+		assertEquals(WorkStatus.FAILED, result.getStatus());
+		assertEquals(MissingParameterException.class, result.getError().getClass());
+		verify(apiInstanceMock, Mockito.times(0)).create(dto);
+	}
+
+	private NotificationMessageCreateRequestDTO buildNotificationMessageCreateRequestDTO(String type, String body,
+			String subject, List<String> usernames, List<String> groupnames
+
+	) {
+		NotificationMessageCreateRequestDTO notificationMessageCreateRequestDTO = new NotificationMessageCreateRequestDTO();
+
+		notificationMessageCreateRequestDTO.messageType(type);
+		notificationMessageCreateRequestDTO.body(body);
+		notificationMessageCreateRequestDTO.subject(subject);
+		notificationMessageCreateRequestDTO.usernames(usernames);
+		notificationMessageCreateRequestDTO.groupnames(groupnames);
+
+		return notificationMessageCreateRequestDTO;
+	}
+
+	private void putParamsToCtx(NotificationMessageCreateRequestDTO dto, WorkContext ctx) {
+		HashMap<String, String> map = new HashMap<>();
+
+		putInMap(map, "type", dto.getMessageType());
+		putInMap(map, "body", dto.getBody());
+		putInMap(map, "subject", dto.getSubject());
+		putInMap(map, "userNames", listToString(dto.getUsernames()));
+		putInMap(map, "groupNames", listToString(dto.getGroupnames()));
+
+		WorkContextDelegate.write(ctx, WorkContextDelegate.ProcessType.WORKFLOW_TASK_EXECUTION, underTest.getName(),
+				WorkContextDelegate.Resource.ARGUMENTS, map);
+	}
+
+	private void putInMap(HashMap<String, String> map, String key, String value) {
+		if (value != null) {
+			map.put(key, value);
+		}
+	}
+
+	private String listToString(List<String> usernames) {
+		if (usernames == null) {
+			return null;
+		}
+		String result = "";
+		for (int i = 0; i < usernames.size(); i++) {
+			if (i > 0) {
+				result = result + ";";
+			}
+			result = result + usernames.get(i);
+		}
+		return result;
```

Comment:
Can this be simplified with `String.join(";", usernames);` ?

Reply:
Changed, thanks for the tip :-)